### PR TITLE
Fetch keys by a given Pattern

### PR DESCRIPTION
**Pull request by Sam-Serpoosh**
There is a document shared with you which contains detailed explanation. But here's a **TL;DR**:

In order to do some caching and performance improvements on the client side, it would be a good idea to have an API call on this library which gets a status and returns ALL the **keys** with that pattern.

### CODE REVIEW

**Comment by abedra** [comment]
These look good, thanks! One last question, did you run checkstyle to make sure there weren't any offenses?

File: src/main/java/com/repsheet/librepsheet/CIDRUtils.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.repsheet.librepsheet;
+
+import com.aaronbedra.orchard.CIDR;
+import com.aaronbedra.orchard.OrchardException;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+
+/**
+ * Utility class for working with Repsheet CIRD blocks.
```
Comment (simonwhitear) [inline src/main/java/com/repsheet/librepsheet/CIDRUtils.java]: CIDR not CIRD.

Comment (abedra) [inline src/main/java/com/repsheet/librepsheet/CIDRUtils.java]: CIDR blocks are not a concept unique to Repsheet either. It should just be CIDR blocks.

Comment (Sam-Serpoosh) [inline src/main/java/com/repsheet/librepsheet/CIDRUtils.java]: Oops! Thanks, fixed!

**Comment by abedra** [comment]
Thanks!

File: src/main/java/com/repsheet/librepsheet/Connection.java [hunk 2]
```diff
@@ -58,6 +59,18 @@ public final Actor lookup(final Actor.Type type, final String value) {
         return new Actor(type, value, Actor.Status.OK, null);
     }
 
+    public final Set<String> lookupByStatus(final Actor.Status status) {
```
Comment (davidhanley) [inline src/main/java/com/repsheet/librepsheet/Connection.java]: I would have liked if this was called getAllKeysWithStatus

**Pull request merged**